compiler/structs.jou

```diff
@@ -87,12 +86,29 @@ class GlobalVariable:
     defined_in_current_file: bool  # not declare-only (e.g. stdout) or imported
     usedptr: bool*  # If non-NULL, set to true when the variable is used. This is how we detect unused imports.
 
+
 class LocalVariable:
-    id: int  # Unique, but you can also compare pointers to Variable.
+    id: int  # Unique, but you can also compare pointers to LocalVariable.
     name: byte[100]  # Same name as in user's code, empty for temporary variables created by compiler
     type: Type*
     is_argument: bool    # First n variables are always the arguments
 
+    def print(self) -> int:
+        if self->name[0] != '\0':
+            printf("%s", self->name)
+        else:
+            # Anonymous temporary variable created by compiler.
+            printf("$%d", self->id)
+
+    # Return how many characters print() outputs.
+    def print_width(self) -> int:
```

Comment:
Instead of returning width, would be better to just print the variable to the correct width here, and `printf()` makes that sorta thing quite easy.